src/nspawn/nspawn.c

```diff
@@ -2725,6 +2753,33 @@ static int inner_child(
         return log_error_errno(r, "execv() failed: %m");
 }
 
+static int setup_sd_notify_child(void) {
+        static const int one = 1;
+        int fd;
+        union sockaddr_union sa = {
+                .sa.sa_family = AF_UNIX,
+        };
+        int r;
+
+        fd = socket(AF_UNIX, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0);
+        if (fd < 0)
+                return log_error_errno(errno, "Failed to allocate notification socket: %m");
+
+        (void) mkdir_parents_label(nspawn_notify_socket_path, 0755);
```

Comment:
thinking about this, I think using the non-selinux version of the call is more appropriate here, after all this is run inside the container namespace already, where selinux does not exist. Or in other words, please use `mkdir_parents()` instead of `mkdir_parents_label()` here.